### PR TITLE
[BCF-3381] - Implement changeset LatestHead

### PR DESCRIPTION
**Pull request by ilija42**
core ref: 9eb801069a307d5495036f8f716ff277db95b4be
- [common PR](https://github.com/smartcontractkit/chainlink-common/pull/760)
- [core PR](https://github.com/smartcontractkit/chainlink/pull/14394)
- [cosmos PR](https://github.com/smartcontractkit/chainlink-cosmos/pull/419)
- [solana PR](https://github.com/smartcontractkit/chainlink-solana/pull/850)
- [chainlink-ccip](https://github.com/smartcontractkit/chainlink-ccip/pull/116)

### CODE REVIEW

**Review by Farber98 — DISMISSED** [review]

**Review by kidambisrinivas — DISMISSED** [review]

**Review by kidambisrinivas — DISMISSED** [review]

**Review by kidambisrinivas — DISMISSED** [review]

**Review by Farber98 — DISMISSED** [review]

**Comment by augustbleeds** [comment]
@ilija42 when will the corresponding changes be pushed to core (chainlink)? We are actively going to be pushing new contracts and modifying the e2e tests so I want to make sure that the e2e tests are green within 1 day of this PR being merged in.

**Review by augustbleeds — CHANGES_REQUESTED** [review]
@ilija42 changes look good. but it's hard to know if this will break e2e tests since you're introducing some changes in the integration tests folder.

when will the corresponding relayer changes be pushed to core (chainlink)? We are actively going to be pushing new contracts and modifying the e2e tests this sprint so I want to make sure that the e2e tests are green within 1 day of this PR being merged in.

File: integration-tests/common/test_common.go [hunk 1]
```diff
@@ -205,8 +205,8 @@ func (m *OCRv2TestState) DeployCluster() {
 		m.Clients.ChainlinkClient.NKeys, m.TestConfig.err = m.Common.CreateNodeKeysBundle(m.Clients.ChainlinkClient.ChainlinkNodes)
 		require.NoError(m.TestConfig.T, m.TestConfig.err)
 	} else {
-		m.Clients.ChainlinkClient.ChainlinkNodes = m.Clients.DockerEnv.ClCluster.NodeAPIs()
-		m.Clients.ChainlinkClient.NKeys, m.TestConfig.err = m.Common.CreateNodeKeysBundle(m.Clients.DockerEnv.ClCluster.NodeAPIs())
+		m.Clients.ChainlinkClient.ChainlinkNodes = m.Clients.DockerEnv.CLClusterTestEnv.ClCluster.NodeAPIs()
```
Comment (augustbleeds) [inline integration-tests/common/test_common.go:208]: this is necessary because chainlink v2 module has been updated?

Comment (ilija42) [reply integration-tests/common/test_common.go:208]: Yeah, @smickovskid fixed this

**Comment by ilija42** [comment]
> @ilija42 changes look good. but it's hard to know if this will break e2e tests since you're introducing some changes in the integration tests folder.
> 
> when will the corresponding relayer changes be pushed to core (chainlink)? We are actively going to be pushing new contracts and modifying the e2e tests this sprint so I want to make sure that the e2e tests are green within 1 day of this PR being merged in.

Starknet tests were passing in the chainlink-common Starknet integration test, should be the same thing? All of the changes in core will be merged within an hour.

File: integration-tests/common/test_common.go (hunk 1 above)
Comment (augustbleeds) [reply integration-tests/common/test_common.go:208]: Oh ok awesome!

**Comment by augustbleeds** [comment]
> > @ilija42 changes look good. but it's hard to know if this will break e2e tests since you're introducing some changes in the integration tests folder.
> > when will the corresponding relayer changes be pushed to core (chainlink)? We are actively going to be pushing new contracts and modifying the e2e tests this sprint so I want to make sure that the e2e tests are green within 1 day of this PR being merged in.
> 
> Starknet tests were passing in the chainlink-common Starknet integration test, should be the same thing? All of the changes in core will be merged within an hour.

Thanks! I'll add the ability to test the custom core reference https://github.com/smartcontractkit/chainlink-starknet/pull/524 and then we can retest

**Comment by augustbleeds** [comment]
yay it's working -- i love seeing the ✅ .

**Review by aalu1418 — APPROVED** [review]

**Review by augustbleeds — APPROVED** [review]

**Pull request merged**